arch/arc/core/arc_connect.c

```diff
@@ -26,8 +26,15 @@ static struct k_spinlock arc_connect_spinlock;
 /* Generate an inter-core interrupt to the target core */
 void z_arc_connect_ici_generate(u32_t core)
 {
+	u32_t ipi_was_pending;
+
 	LOCKED(&arc_connect_spinlock) {
-		z_arc_connect_cmd(ARC_CONNECT_CMD_INTRPT_GENERATE_IRQ, core);
+		z_arc_connect_cmd(ARC_CONNECT_CMD_INTRPT_READ_STATUS, core);
```

Comment:
So now if for whatever reason other core is still busy processing another cross-core IRQ it won't get a new one and so effectively we'll forget to deliver some important piece of information to the remote core.

In that case we need to have a loop in which we'll be waiting until previous IRQ gets acknowledged.

Reply:
I refer arc codes in linux to do this change.  It make sense to avoid uncessary operation of arc connect.  As the ipi from initiator to the receiver is already pending and consider current Zephyr's codes,  nothing will be missed.

Reply:
I remove the change. it has a potential bug mentioned by @andyross